src/Sentry/SentryFailedRequestHandler.cs

```diff
@@ -0,0 +1,100 @@
+using System.Net;
+using System.Reflection.PortableExecutable;
+using Sentry.Internal;
+using Sentry.Protocol;
+
+namespace Sentry;
+
+internal class SentryFailedRequestHandler : ISentryFailedRequestHandler
+{
+    private readonly IHub _hub;
+    private readonly SentryOptions _options;
+
+    public const string MechanismType = "SentryFailedRequestHandler";
+
+    internal SentryFailedRequestHandler(IHub hub, SentryOptions options)
+    {
+        _hub = hub;
+        _options = options;
+    }
+
+    public void HandleResponse(HttpResponseMessage response)
+    {
+        // Ensure reponse and request are not null
+        if (response?.RequestMessage is null)
+        {
+            return;
+        }
+
+        // Don't capture if the option is disabled
+        if (_options?.CaptureFailedRequests is false)
+        {
+            return;
+        }
+
+        // Don't capture events for successful requets
+        if (_options?.FailedRequestStatusCodes.Any(range => range.Contains(response.StatusCode)) is false)
+        {
+            return;
+        }
+
+        // Ignore requests to the Sentry DSN
+        var uri = response.RequestMessage.RequestUri;
+        if (_options?.Dsn is { } dsn && new Uri(dsn).Host.Equals(uri?.Host, StringComparison.OrdinalIgnoreCase))
+        {
+            return;
+        }
+
+        // Ignore requests that don't match the FailedRequestTargets
+        var requestString = uri?.OriginalString ?? "";
+        if (_options?.FailedRequestTargets.ContainsMatch(requestString) is false)
+        {
+            return;
+        }
+
+        // Capture the event
+        try
+        {
+            response.EnsureSuccessStatusCode();
+        }
+        catch (HttpRequestException exception)
+        {
+            exception.SetSentryMechanism(MechanismType);
+
+            var @event = new SentryEvent(exception);
+
+            var sentryRequest = new Request
+            {
+                Url = uri?.AbsoluteUri,
+                QueryString = uri?.Query,
+                Method = response.RequestMessage.Method.Method,
+            };
+            if (_options?.SendDefaultPii is true)
+            {
+                sentryRequest.Cookies = response.RequestMessage.Headers.GetCookies();
+                sentryRequest.AddHeaders(response.RequestMessage.Headers);
+            }
+
+            var responseContext = new Response {
+                StatusCode = (short)response.StatusCode
+            };
+
+            // .NET 4.8 doesn't set the Content-Length header
+            // https://github.com/dotnet/runtime/issues/16162
+#if NET5_0_OR_GREATER
+            responseContext.BodySize = response.Content?.Headers?.ContentLength;
+#endif
```

Comment:
Despite their bug, we probably shouldn't restrict in this way.  If it returns `null`, there's no harm to us - as that means we couldn't collect it.  (If they fix this bug, we'd not get that benefit.)

Also, this particular conditional compilation constant would exclude .NET Standard and earlier .NET core targets as well.

So unless it is throwing an exception, then just let it return null.  Also the extra null checks aren't required.

```suggestion
            responseContext.BodySize = response.Content.Headers.ContentLength;
```

Reply:
I couldn't reproduce that bug anyway.  Perhaps they've fixed it and forgot to update GitHub.  Either way, let's just take the value or null.